interchain-token-service/src/events.rs

```diff
@@ -0,0 +1,103 @@
+use cosmwasm_std::{Attribute, Event};
+use router_api::{Address, ChainName};
+
+use crate::primitives::ItsMessage;
+
+pub enum ItsContractEvent {
+    ItsMessageReceived {
+        source_chain: ChainName,
+        destination_chain: ChainName,
+        message: ItsMessage,
+    },
+    ItsAddressSet {
+        chain: ChainName,
+        address: Address,
+    },
+    ItsAddressRemoved {
+        chain: ChainName,
+    },
+}
+
+impl From<ItsContractEvent> for Event {
+    fn from(event: ItsContractEvent) -> Self {
+        match event {
+            ItsContractEvent::ItsMessageReceived {
+                source_chain,
+                destination_chain,
+                message,
+            } => make_its_message_event(
+                "its_message_received",
+                source_chain,
+                destination_chain,
+                message,
+            ),
+            ItsContractEvent::ItsAddressSet { chain, address } => {
+                Event::new("trusted_address_updated")
+                    .add_attribute("chain", chain.to_string())
+                    .add_attribute("address", address.to_string())
+            }
+            ItsContractEvent::ItsAddressRemoved { chain } => {
+                Event::new("trusted_address_removed").add_attribute("chain", chain.to_string())
+            }
+        }
+    }
+}
+
+fn make_its_message_event(
+    event_name: &str,
+    source_chain: ChainName,
+    destination_chain: ChainName,
+    msg: ItsMessage,
+) -> Event {
+    let mut attrs = vec![
+        Attribute::new("source_chain", source_chain.to_string()),
+        Attribute::new("destination_chain", destination_chain.to_string()),
+        Attribute::new("message_type", format!("{:?}", msg)),
```

Comment:
Does this not display the whole msg?

Reply:
good catch, fixed. Thoughts about using a string vs a number for the event. String representation is not great if we rename something, but it's more readable